### PR TITLE
KOGITO-5554: Data-index API connection: getTasksSchema, TaskUpdate, T…

### PR DESCRIPTION
**Pull request by nmirasch**
…askPartialUpdate

jira: https://issues.redhat.com/browse/KOGITO-5554 

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

File: data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls [hunk 1]
```diff
@@ -292,6 +294,7 @@ type UserTaskInstance {
     referenceName: String
     lastUpdate: DateTime!
     endpoint: String
+    schema(user: String, groups:[String]): String
```
Comment (cristianonicolai) [inline data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls:297]: nice one! :)

File: apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/src/main/java/org/kie/kogito/index/AbstractProcessDataIndexIT.java [hunk 2]
```diff
@@ -239,6 +243,75 @@ public void testProcessInstanceEvents() {
...
+                    .body("{ \"query\" : \"{ UserTaskInstances (where: {id: {equal:\\\"" + taskId + "\\\" }}){ " +
+                            "schema ( user: \\\"manager\\\", groups: [\\\"managers\\\", \\\"users\\\", \\\"IT\\\"] )" +
+                            "}}\"}")
+                    .when().post("/graphql")
+                    .then()
+                    .statusCode(200)
+                    .extract().path("data.UserTaskInstances[0].schema");
+
+            ObjectMapper mapper = new ObjectMapper();
+            assertEquals(mapper.readTree(taskSchema), mapper.readTree(expectedSchema));
```
Comment (cristianonicolai) [inline apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/src/main/java/org/kie/kogito/index/AbstractProcessDataIndexIT.java]: maybe use _assertThatJson_ ? similar to https://github.com/kiegroup/kogito-apps/blob/main/data-index/data-index-service/data-index-service-common/src/test/java/org/kie/kogito/index/json/ProcessInstanceMetaMapperTest.java#L44

Comment (nmirasch) [reply apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-common/src/main/java/org/kie/kogito/index/AbstractProcessDataIndexIT.java]: @cristianonicolai I have changed the test to check individually  the expected  json values, the same as is used at assertThatJson. Could you take a look and see if this solution works?

**Review by cristianonicolai — APPROVED** [review]

File: data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls [hunk 3]
```diff
@@ -22,6 +22,8 @@ type Mutation {
     NodeInstanceCancel(id:String, nodeInstanceId:String): String
     JobCancel(id: String): String
     JobReschedule(id: String, data: String): String
+    TaskUpdate(id: String,taskId: String, user:String, groups:[String], taskInfo: UserTaskInfoArgument): String
+    TaskPartialUpdate(id: String,taskId: String, user:String, groups:[String], taskInfo: UserTaskInfoArgument): String
```
Comment (radtriste) [inline data-index/data-index-service/data-index-service-common/src/main/resources/schema.graphqls]:
```suggestion
    TaskUpdate(id: String, taskId: String, user:String, groups:[String], taskInfo: UserTaskInfoArgument): String
    TaskPartialUpdate(id: String, taskId: String, user:String, groups:[String], taskInfo: UserTaskInfoArgument): String
```

**Review by radtriste — APPROVED** [review]
lgtm. typo comment

**Pull request merged**